Purchases/RCCrossPlatformSupport.h

```diff
@@ -6,19 +6,55 @@
 //  Copyright © 2019 RevenueCat. All rights reserved.
 //
 
-#if TARGET_OS_IPHONE
-#import <UIKit/UIKit.h>
+#if TARGET_OS_IOS || TARGET_OS_TV || TARGET_OS_MACCATALYST
 #define APP_DID_BECOME_ACTIVE_NOTIFICATION_NAME UIApplicationDidBecomeActiveNotification
 #define APP_WILL_RESIGN_ACTIVE_NOTIFICATION_NAME UIApplicationWillResignActiveNotification
-#elif TARGET_OS_MAC
-#import <AppKit/AppKit.h>
+#elif TARGET_OS_OSX
 #define APP_DID_BECOME_ACTIVE_NOTIFICATION_NAME NSApplicationDidBecomeActiveNotification
 #define APP_WILL_RESIGN_ACTIVE_NOTIFICATION_NAME NSApplicationWillResignActiveNotification
+#elif TARGET_OS_WATCH
+#define APP_DID_BECOME_ACTIVE_NOTIFICATION_NAME NSExtensionHostDidBecomeActiveNotification
+#define APP_WILL_RESIGN_ACTIVE_NOTIFICATION_NAME NSExtensionHostWillResignActiveNotification
+#endif
+
+#if TARGET_OS_IPHONE
+#import <UIKit/UIKit.h>
+#elif TARGET_OS_OSX
+#import <AppKit/AppKit.h>
 #endif
+
 #if TARGET_OS_MACCATALYST
 #define PLATFORM_HEADER @"uikitformac"
-#elif TARGET_OS_IPHONE
+#elif TARGET_OS_IOS
 #define PLATFORM_HEADER @"iOS"
-#elif TARGET_OS_MAC
+#elif TARGET_OS_OSX
 #define PLATFORM_HEADER @"macOS"
+#elif TARGET_OS_WATCH
+#define PLATFORM_HEADER @"watchOS"
+#elif TARGET_OS_TV
+#define PLATFORM_HEADER @"tvOS"
+#endif
+
+// Should match available platforms in
+// https://developer.apple.com/documentation/uikit/uidevice?language=objc
+#if TARGET_OS_IOS || TARGET_OS_TV || TARGET_OS_MACCATALYST
+#define UI_DEVICE_AVAILABLE 1
+#else
+#define UI_DEVICE_AVAILABLE 0
+#endif
+
+// Should match available platforms in
+// https://developer.apple.com/documentation/iad/adclient?language=objc
+#if TARGET_OS_IOS || TARGET_OS_MACCATALYST
+#define AD_CLIENT_AVAILABLE 1
+#else
+#define AD_CLIENT_AVAILABLE 0
+#endif
+
+// Should match available platforms in
+// https://developer.apple.com/documentation/storekit/skpaymenttransactionobserver/2877502-paymentqueue?language=objc
+#if TARGET_OS_IOS || TARGET_OS_MACCATALYST
```

Comment:
shouldn't this be `TARGET_OS_IOS || TARGET_OS_TV`?

Reply:
great catch. I'm glad I added the link there, I'm not glad that I didn't check in more detail XD